### PR TITLE
Implementation of log.go

### PR DESCRIPTION
**Pull request by ian-kent**
### What

Implemented logger for go which matches logging standards.

### How to review

* Check README makes sense
* Check godoc makes sense
* Check tests are useful, pass, and don't have any obvious gaps
* Check the output matches the spec
* Review the implementation code
* Have a look at ONSdigital/florence#330 to see how its used
* Try using it!

### Who can review

Anyone except @ian-kent

### CODE REVIEW

File: log/auth.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	IdentityType identityType `json:"identity_type,omitempty"`
+}
+
+type identityType string
+
+const (
+	// SERVICE represents a service account type
+	SERVICE identityType = "service"
+	// USER represents a user account type
+	USER = "user"
```
Comment (mikeAdamss) [inline log/auth.go]: should this have an identity type as well?

[the remark this answers is not in the document]
Comment (ian-kent) [reply log/auth.go]: 👍

File: log/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+func TestAuth(t *testing.T) {
+	Convey("Auth function returns a *eventAuth", t, func() {
+		auth := Auth(USER, "test")
+		So(auth, ShouldHaveSameTypeAs, &eventAuth{})
+		So(auth, ShouldImplement, (*option)(nil))
+
+		Convey("*eventAuth has the correct fields", func() {
+			ea := Auth(USER, "test1").(*eventAuth)
+			So(ea.IdentityType, ShouldEqual, USER)
```
[the remark this answers is not in the document]
Comment (ian-kent) [reply log/auth_test.go]: hm - I'm confused 😕 either the type definition isn't working, Auth() shouldn't have accepted it, or Go has implicitly converted the string to an identityType

I'm not sure which it is, but I think fixing the const should fix all of those possibilities? 🤔

[the remark this answers is not in the document]
Comment (ian-kent) [reply log/auth_test.go]: (having said that - this test only cares that the value passed in to Auth() is the same as the one in IdentityType - the actual types don't matter, just that the value is taken from the args and stored on the struct - so I think the test is right even if I'm confused a bit by the implementation!)

**Pull request merged**